base/shell/explorer/taskswnd.cpp

```diff
@@ -1736,7 +2488,148 @@ class CTaskSwitchWnd :
         }
         else if (TaskGroup != NULL)
         {
-            /* FIXME: Implement painting for task groups */
+            Ret = CDRF_SKIPDEFAULT;
+            /* Make the entire button flashing if necessary */
+            if (nmtbcd->nmcd.uItemState & CDIS_MARKED)
+            {
+                if (!m_Theme)
+                {
+                    SelectObject(nmtbcd->nmcd.hdc, GetSysColorBrush(COLOR_HIGHLIGHT));
+                    Rectangle(nmtbcd->nmcd.hdc,
+                        nmtbcd->nmcd.rc.left,
+                        nmtbcd->nmcd.rc.top,
+                        nmtbcd->nmcd.rc.right,
+                        nmtbcd->nmcd.rc.bottom);
+                }
+                else
+                {
+                    DrawThemeBackground(m_Theme, nmtbcd->nmcd.hdc, TDP_FLASHBUTTONGROUPMENU, 0, &nmtbcd->nmcd.rc, 0);
+                }
+                nmtbcd->clrText = GetSysColor(COLOR_HIGHLIGHTTEXT);
+            }
+
+            WCHAR buttonText[256];
+
+            TBBUTTONINFOW btni;
+            btni.cbSize = sizeof(TBBUTTONINFOW);
+            btni.dwMask = TBIF_STATE | TBIF_TEXT;
+            btni.cchText = 256;
+            btni.pszText = buttonText;
+            m_TaskBar.GetButtonInfo(TaskGroup->Index, &btni);
+
+            DWORD padding = m_TaskBar.GetPadding();
+            WORD paddingCy = HIWORD(padding);
+            WORD paddingCx = LOWORD(padding);
+            INT bitmapWidth;
+            INT bitmapHeight;
+
+            ImageList_GetIconSize(m_ImageList, &bitmapWidth, &bitmapHeight);
+
+            if (!m_Theme)
+            {
+                if (nmtbcd->nmcd.uItemState & (CDIS_SELECTED | CDIS_CHECKED))
+                    DrawEdge (nmtbcd->nmcd.hdc, &nmtbcd->nmcd.rc, EDGE_SUNKEN, BF_RECT | BF_MIDDLE);
+                else
+                    DrawEdge (nmtbcd->nmcd.hdc, &nmtbcd->nmcd.rc, EDGE_RAISED,
+                    BF_SOFT | BF_RECT | BF_MIDDLE);
+            }
+
+            RECT rcText = nmtbcd->nmcd.rc;
+            RECT rcBitmap = nmtbcd->nmcd.rc;
+            RECT rcArrow = nmtbcd->nmcd.rc;
+            InflateRect(&rcText, -GetSystemMetrics(SM_CXEDGE), 0);
+            rcText.left += bitmapWidth + 6; //TODO Default list gap
+
+            if(btni.fsState & (TBSTATE_PRESSED | TBSTATE_CHECKED))
+                OffsetRect(&rcText, 1, 1);
+
+            rcBitmap.left += GetSystemMetrics(SM_CXEDGE) + paddingCx / 2;
+            rcBitmap.top += paddingCy / 2;
+
+            if (nmtbcd->nmcd.uItemState & (CDIS_SELECTED | CDIS_CHECKED))
+            {
+                rcBitmap.left++;
+                rcBitmap.top++;
+            }
+
+            rcArrow.left = max(rcArrow.left, rcArrow.right - 13);
```

Comment:
What's this "13" about?

Reply:
I think is a prepcalculated x offset, i could try to add the formula there, but u think is always a constant and always the same

Reply:
You could add a `#define` for this that explains to calculation.